test/decl/func/dynamic_self.swift

```diff
@@ -412,4 +412,88 @@ class SelfOperator {
 func useSelfOperator() {
   let s = SelfOperator()
   _ = s + s
-}
\ No newline at end of file
+}
+
+// These references to Self are now possible
```

Comment:
Mention SE-0068 in the comment. Also I think these tests belong more in test/type/self.swift and not test/decl/func/dynamic_self.swift.

Reply:
moved tests and added reference to SE-0068